elliott/elliottlib/cli/__main__.py

```diff
@@ -202,77 +203,64 @@ async def verify_payload(runtime, payload, advisory):
-                'payload': vr,
-                'errata': all_advisory_nvrs[image]
-            }
-
-    if missing_in_errata:  # check if missing images are already shipped or pending to ship
-        advisory_nvrs: Dict[int, List[str]] = {}  # a dict mapping advisory numbers to lists of NVRs
-        green_print(f"Checking if {len(missing_in_errata)} missing images are shipped...")
-        for nvr in missing_in_errata.copy().values():
-            # get the list of advisories that this build has been attached to
-            build = elliottlib.errata.get_brew_build(nvr)
-            # filter out dropped advisories
-            advisories = [ad for ad in build.all_errata if ad["status"] != "DROPPED_NO_SHIP"]
-            if not advisories:
-                red_print(f"Build {nvr} is not attached to any advisories.")
+    # filter out rhcos nvrs since they are special and not auto-stitched and fetched by util.get_nvrs_from_payload
+    all_advisory_nvrs = [n for n in all_advisory_nvrs if "rhcos-" not in n[0]]
+
+    all_advisory_nvrs = set(all_advisory_nvrs)
+    runtime.logger.info(f"Found {len(all_advisory_nvrs)} nvrs in advisory")
+
+    missing_in_advisory = all_payload_nvrs - all_advisory_nvrs
+    extra_in_advisory = all_advisory_nvrs - all_payload_nvrs
+    runtime.logger.info(f"Extra in advisory {len(extra_in_advisory)}")
+
+    in_pending_advisory = set()
+    in_shipped_advisory = set()
+    attached_but_shipped_or_pending = set()
+
+    runtime.logger.info(f"Checking if any payload nvrs that are attached have been shipped...")
+    for nvr_tuple in all_payload_nvrs & all_advisory_nvrs:
+        nvr = "-".join(nvr_tuple)
+        build = elliottlib.errata.get_brew_build(nvr)
+        for ad in build.all_errata:
+            if ad['id'] == advisory:
                 continue
-            for advisory in advisories:
-                if advisory["status"] == "SHIPPED_LIVE":
-                    green_print(f"Missing build {nvr} has been shipped with advisory {advisory}.")
-                else:
-                    yellow_print(f"Missing build {nvr} is in another pending advisory.")
-                advisory_nvrs.setdefault(advisory["id"], []).append(nvr)
-            name = nvr.rsplit("-", 2)[0]
-            del missing_in_errata[name]
-        if advisory_nvrs:
-            click.echo(f"Getting information of {len(advisory_nvrs)} advisories...")
-            for advisory, nvrs in advisory_nvrs.items():
-                advisory_obj = elliottlib.errata.get_raw_erratum(advisory)
-                adv_type, adv_info = next(iter(advisory_obj["errata"].items()))
-                item = {
-                    "id": advisory,
-                    "type": adv_type.upper(),
-                    "url": elliottlib.constants.errata_url + f"/{advisory}",
-                    "summary": adv_info["synopsis"],
-                    "state": adv_info["status"],
-                    "nvrs": nvrs,
-                }
```

Comment:
Wasn't apparent to me why we were fetching this information. Afaik it is not being used by anything and isn't super useful